src/Application/Controller/XMLRPC/Handler.php

```diff
@@ -376,22 +376,29 @@ public function createEncodingTicket($meta_ticket_id, $encoding_profile_id, arra
 				throw new Exception(__FUNCTION__ . ': given encoding profile not available in this project', 1204);
 			}
 			
+			$initial_state = $project->queryFirstState('encoding')->first();
+
 			try {
-				// create missing encoding ticket
-				$ticket_id = Ticket::createMissingEncodingTicket($meta_ticket_id, $encoding_profile_id);
-				
-				Log::warning('ticket_id: '.$ticket_id);
+				$ticket_data = [
+					'parent_id' => $metaTicket['id'],
+					'project_id' => $metaTicket['project_id'],
+					'fahrplan_id' => $metaTicket['fahrplan_id'],
+					'priority' => $encodingProfileVersion['priority'],
+					'ticket_type' => 'encoding',
+					'ticket_state' => $initial_state['ticket_state'],
+					'encoding_profile_version_id' => $encodingProfileVersion['id']
+				];
 				
-				// get ticket
-				$ticket = Ticket::find(['id' => $ticket_id]);
+				// create encoding ticket
+				$ticket = Ticket::createOrThrow($ticket_data);
```

Comment:
When `$ticket_data` is not needed anywhere else I would probably prefer using an inline array as argument.

Reply:
Sure.